lib/matplotlib/widgets.py

```diff
@@ -2006,6 +2007,42 @@ def set_visible(self, visible):
         for artist in self.artists:
             artist.set_visible(visible)
 
+    def add_artist(self, artist):
+        """Add artist to the selector."""
+        self._artists.append(artist)
+
+    @property
+    def artists(self):
+        """Tuple of the artists of the selector."""
+        return tuple(self._artists)
+
+    @property
+    def _selection_artist(self):
+        """The selection artist."""
+        return self.artists[0]
+
+    def set_props(self, **props):
+        """
+        Set the properties of the selection artist. See the `props` argument
+        in the selector docstring to know which properties are supported.
+        """
+        self._selection_artist.set(**props)
+        self.update()
```

Comment:
Is it really necessary to call `update()` here? That means e.g. if the user calls `set_props()` and `set_handle_props()` that there are two redraws.

If we think the update *is* neccessary, these functions are the way to go. If not, it might be simpler to just expose the `selection_artist` and `handle_artists` properties and let the user do with them whatever they feel necessary.

Reply:
`update` is necessary when using blit. Without calling `update`, we get the following behaviour:
- `useblit=True`, the artists are not updated
- `useblit=False`, the artists are updated

Using `selection_artist` and `handle_artists` will be more verbose, not only because `update` will need to be called when using blit but also because `handle_artists` will be a tuple, which needs to be iterated over. However, as you said, using `selection_artist` and `handle_artists` is more transparent and simple and I guess users changing the properties on the fly will most likely be experienced user (if there are using blit, for example), i. e. they should be fine with this level of verbosity and prefer transparency!